File: backend/app/schemas/perfil.py
from typing import Optional
from pydantic import BaseModel, ConfigDict

class PerfilBase(BaseModel):
    nombre_usuario: str
    telefono: Optional[str] = None
    foto_perfil: Optional[str] = None
    descripcion: Optional[str] = None
    notif_activas: bool = True
    limite_cronometro: Optional[int] = None

class PerfilCreate(PerfilBase):
    id_usuario: int

class PerfilUpdate(BaseModel):
    nombre_usuario: Optional[str] = None
    telefono: Optional[str] = None
    foto_perfil: Optional[str] = None
    descripcion: Optional[str] = None
    notif_activas: Optional[bool] = None
    limite_cronometro: Optional[int] = None

class PerfilOut(PerfilBase):
    id_perfil: int
    id_usuario: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/services/perfil.ts
// src/services/perfil.ts
import { apiFetch } from "./api";

export interface PerfilBackend {
  id_perfil: number;
  nombre_usuario: string;
  telefono: string | null;
  foto_perfil: string | null;
  descripcion: string | null;
  notif_activas: boolean;
  limite_cronometro: number | null;
  id_usuario: number;
}

// Todos opcionales porque el PerfilUpdate del backend acepta solo lo que cambió
export interface ActualizarPerfilPayload {
  nombre_usuario?: string;
  telefono?: string;
  foto_perfil?: string;
  descripcion?: string;
  notif_activas?: boolean;
  limite_cronometro?: number;
}

export function obtenerMiPerfil(): Promise<PerfilBackend> {
  return apiFetch<PerfilBackend>("/perfiles/me");
}

export function actualizarMiPerfil(
  datos: ActualizarPerfilPayload
): Promise<PerfilBackend> {
  return apiFetch<PerfilBackend>("/perfiles/me", {
    method: "PATCH",
    body: JSON.stringify(datos),
  });
}